### PR TITLE
Add a config Class

### PR DESCRIPTION
**Pull request by AntoineJac**
Similar to the setConfig for Prebid.js create a class for current and future feature

### CODE REVIEW

File: sdk/PrebidServerAdapter/PrebidServerModule/PBServerRequestBuilder.m [hunk 1]
```diff
@@ -87,6 +88,16 @@ - (NSDictionary *)openrtbRequestExtension
     NSMutableDictionary *requestPrebidExt = [[NSMutableDictionary alloc] init];
     requestPrebidExt[@"targeting"] = @{@"lengthmax" : @(20), @"pricegranularity":@"medium"};
```
Comment (ppuviarasu) [inline sdk/PrebidServerAdapter/PrebidServerModule/PBServerRequestBuilder.m]: We need to remove this

File: sdk/PrebidMobile/PBConfig.m [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
++ (void)resetSharedInstance {
+    onceToken = 0;
+    sharedInstance = nil;
+}
+
+-(void) setPriceGranularity:(PBPriceGranularity2)priceGranularity{
+    _priceGranularity = priceGranularity;
+    NSLog(@"Antoine test Granularity is %ld", (long)_priceGranularity);
```
[the remark this answers is not in the document]
Comment (AntoineJac) [reply sdk/PrebidMobile/PBConfig.m]: Hi @ppuviarasu,

Sorry I had committed the wrong branch. Just update with the last version. THanks,

**Comment by AntoineJac** [comment]
@anwzhang, here is the last version. By default I have followed your recommandation and it will pass the account for the StoredRequestId without priceGranularity.

The StoredRequestId and the PriceGranularity could be set up by calling the PbConfig class methods. 

If all ok can you please merge. Thanks,

**Comment by yoalex5** [comment]
@AntoineJac Could you please review my PR
https://github.com/AntoineJac/prebid-mobile-ios/pull/1

**Comment by AntoineJac** [comment]
@yoalex5 , check, review and confirm I merge your PR. Thanks a lot :)

**Comment by yoalex5** [comment]
It seems good

**Comment by AntoineJac** [comment]
@ppuviarasu , you are right that's why the normal use case is passing the storeRequestId. However I guess we should let them the option to hardcode the PriceGranularity in the app to avoid any mistake change on the server configs.

@bretg (Rubicon Product Manager), what are your thoughts about this? 
I can easily remove the PriceGranularity option from my PR if needed.

**Comment by bretg** [comment]
What we need the SDK to do is support a way for the app to generate an OpenRtb2 block in $.ext.prebid.storedrequest.id

i.e. the SDK doesn't need to actually define the granularity, but needs to be able to point to a stored-request that contains the granularity and/or other top-level items in the openrtb JSON. This stored-request-id of course needs to be something the app can define... it may be a UUID or a custom value like "ACCOUNTID-granularity"

Why? Price granularity needs to be defined in a different place in the OpenRTB PBS JSON. For example:

```
{
  "test":1,
  "tmax":500,
  "id": "test-request-id",
  "app": {
        "bundle": "AppNexus.PrebidMobileDemo",
        "ext": {
            "prebid": {
                "version": "0.2.0",
                "source": "prebid-mobile"
            }
        },
        "publisher": {
            "id": "1001"
        },
        "ver": "1.0"
  },
  "imp": [
    {
      "id": "1",
      "banner": {
         "format": [
            {
                "h": 250,
                "w": 300
            }
        ]
      },
      "ext": {
        "prebid": {
          "storedrequest": {
            "id": "IMPRESSION-LEVEL-STORED-REQUEST-ID"
          }
        }
      },
  ],
  "ext": {   //  the SDK can't currently create this part of the JSON object
    "prebid": {
	"storedrequest": {
          "id": "TOP-LEVEL-STORED-REQUEST-ID"
	}
    }
  }
}
```

**Comment by AntoineJac** [comment]
@bretg , yes that's will be available via the setStoreRequestId() from the class.

I will then remove the option from my PR.

**Comment by AntoineJac** [comment]
@ppuviarasu , I have updated the PR with your remarks. Can you please review the PR?

Thanks,

**Comment by anwzhang** [comment]
Close as duplicate as https://github.com/prebid/prebid-mobile-ios/pull/90

**Comment by anwzhang** [comment]
@AntoineJac do you have example why would a publisher wants to set the storedrequest id through and API other than setting the accound id (a.k.a a stored request id) on the init method? Where would they get the stored request id?

**Comment by AntoineJac** [comment]
+ @bretg, @anwzhang ,
I guess most of publisher will only have one settings that could be stored under the init method accound id (a.k.a a stored request id), which is the default use case in my PR. 

But the api should be available in case a publisher has several apps and use different settings. Also Prebid server is also used for AMP and other channel, maybe we will have in future different DB entry such as: "1001_settings_app1", "1001_settings_app2", "1001_settings_amp1"...

**Comment by mjacobsonny** [comment]
@AntoineJac -- Can I ask that you please open an Intent to Implement issue where we can discuss the API change(s) proposed herein?

In the interim, I am going to close this PR.

**Pull request closed without merging**